src/Core/Services/DeviceCryptoService.cs

```diff
@@ -0,0 +1,85 @@
+﻿
+using System;
+using System.Threading.Tasks;
+using Bit.Core.Models.Domain;
+using Bit.Core.Models.Request;
+
+namespace Bit.Core.Abstractions
+{
+    public class DeviceCryptoService
+    {
+        private readonly IApiService _apiService;
+        private readonly IAppIdService _appIdService;
+        private readonly ICryptoFunctionService _cryptoFunctionService;
+        private readonly ICryptoService _cryptoService;
+        private readonly IStateService _stateService;
+
+        public DeviceCryptoService(
+            IApiService apiService,
+            IAppIdService appIdService,
+            ICryptoFunctionService cryptoFunctionService,
+            ICryptoService cryptoService,
+            IStateService stateService)
+        {
+            _apiService = apiService;
+            _appIdService = appIdService;
+            _cryptoFunctionService = cryptoFunctionService;
+            _cryptoService = cryptoService;
+            _stateService = stateService;
+        }
+
+
+        public async Task<DeviceResponse> TrustDeviceAsync()
```

Comment:
❓ I'm not sure where this will be used, but just wanted to ask: Does this need any special exceptions handling or is it ok to just pass it through the caller as it's now?

Reply:
Not yet, this is a helper PR to unblock other work. We'll iterate further if needed.